sippy-ng/src/component_readiness/ComponentReadiness.js

```diff
@@ -0,0 +1,861 @@
+
+  tmp = groupByParameters.get('baseEndTime')
+  let initBaseEndTime = initialEndTime
+  if (tmp != null) {
+    initBaseEndTime = tmp
+  }
+  const [baseEndTime, setBaseEndTime] = React.useState(initBaseEndTime)
+
+  tmp = groupByParameters.get('sampleStartTime')
+  let initSampleStartTime = initialPrevStartTime
+  if (tmp != null) {
+    initSampleStartTime = tmp
+  }
+  const [sampleStartTime, setSampleStartTime] =
+    React.useState(initSampleStartTime)
+
+  tmp = groupByParameters.get('sampleEndTime')
+  let initSampleEndTime = initialPrevEndTime
+  if (tmp != null) {
+    initSampleEndTime = tmp
+  }
+  const [sampleEndTime, setSampleEndTime] = React.useState(initSampleEndTime)
+
+  const excludeUpgradesList = [
+    'no-upgrade',
+    'none',
+    'upgrade-micro',
+    'upgrade-minor',
+  ]
+
+  tmp = groupByParameters.get('exclude_upgrades')
+  let initExcludeUpgradesList = []
+  if (tmp !== null && tmp !== '') {
+    initExcludeUpgradesList = tmp.split(',')
+  }
+  const [excludeUpgradesCheckedItems, setExcludeUpgradesCheckedItems] =
+    React.useState(initExcludeUpgradesList)
+  //console.log('initExcludeUpgradesList: ', initExcludeUpgradesList)
+  console.log('excludeUpgradesCheckedItems: ', excludeUpgradesCheckedItems)
+
+  const excludeVariantsList = [
+    'assisted',
+    'compact',
+    'fips',
+    'hypershift',
+    'microshift',
+    'osd',
+    'proxy',
+    'rt',
+    'serial',
+    'single-node',
+    'standard',
+    'techpreview',
+  ]
+  tmp = groupByParameters.get('exclude_variants')
+  let initExcludeVariantsList = []
+  if (tmp !== null && tmp !== '') {
+    initExcludeVariantsList = tmp.split(',')
+  }
+  const [excludeVariantsCheckedItems, setExcludeVariantsCheckedItems] =
+    React.useState(initExcludeVariantsList)
+  //console.log('initExcludeVariantsList: ', initExcludeVariantsList)
+  console.log('excludeVariantsCheckedItems: ', excludeVariantsCheckedItems)
+
+  const pageTitle = (
+    <Typography variant="h4" style={{ margin: 20, textAlign: 'center' }}>
+      Component Readiness for {baseRelease}
+    </Typography>
+  )
+
+  const { path, url } = useRouteMatch()
+
+  const [fetchError, setFetchError] = React.useState('')
+  const [isLoaded, setIsLoaded] = React.useState(false)
+  const [data, setData] = React.useState({})
+
+  // This is the table we use when the first page is initially rendered.
+  const initialPageTable = {
+    rows: [
+      {
+        component: 'None',
+        columns: [
+          {
+            empty: 'None',
+            status: 3, // Let's start with success
+          },
+        ],
+      },
+    ],
+  }
+  const noDataTable = {
+    rows: [
+      {
+        component: 'No Data found',
+        columns: [
+          {
+            empty: 'None',
+            status: 3, // Let's start with success
+          },
+        ],
+      },
+    ],
+  }
+  useEffect(() => {
+    document.title = `Sippy > Component Readiness`
```

Comment:
This doesn't need to be here, you also have it duplicated below.  I think that one's fine